packages/xstate-wallet/src/store/memory-store.ts

```diff
@@ -177,24 +194,34 @@ export class MemoryStore implements Store {
     this._eventEmitter.emit('addToOutbox', {signedStates: [signedState]});
   }
 
+  async addState(state: SignedState) {
+    const channelId = calculateChannelId(state);
+    const channelStorage = this._channels[channelId] || (await this.initializeChannel(state));
+
+    channelStorage.addState(state, state.signature);
+  }
+
   public getAddress(): string {
     return Object.keys(this._privateKeys)[0];
   }
 
-  pushMessage(message: Message) {
+  async pushMessage(message: Message) {
     const {signedStates, objectives} = message;
 
     if (signedStates) {
       // todo: check sig
       // todo: check the channel involves me
-      signedStates.forEach(signedState => {
-        const {signature, ...state} = signedState;
-        this._eventEmitter.emit('stateReceived', state);
-      });
+      await Promise.all(
+        signedStates.map(async signedState => {
+          const {signature, ...state} = signedState;
+          await this.addState(signedState);
+          this._eventEmitter.emit('stateReceived', state);
```

Comment:
I think this should be switched to `channelUpdated` and we should just use the `ChannelUpdatedFeed` for now.

Reply:
Agreed

Reply:
I would prefer deferring this to a different PR though -- even though `memory-store.test` is testing `stateReceivedFeed` and not `channelUpdatedFeed`, the `stateReceivedFeed` is not on the `Store` interface. So, protocols should not be using it.